src/apis/api.js

```diff
@@ -0,0 +1,19 @@
+import axios from "axios";
+
+const apiKey = import.meta.env.VITE_API_KEY;
+
+const instance = axios.create({
+  baseURL: `https://newsapi.org/v2`,
+  headers: {
+    "Content-Type": "application/json",
+  },
+});
+
+export const fetchCategoryNews = async (category, currentPage, searchKeyword) => {
+  let url = `/top-headlines?country=us&pageSize=15&page=${currentPage}&q=${searchKeyword}&apiKey=${apiKey}`;
+  if (category) {
+    url = `/top-headlines?category=${category}&pageSize=15&page=${currentPage}&q=${searchKeyword}&apiKey=${apiKey}`;
+  }
+  const response = await instance.get(url);
+  return response;
+};
```

Comment:
함수가 하는 일이 명확해서 좋은 것 같습니다. 👍🏻 하지만 개선점이 남아있는거 같습니다. 내용은 URL의 중복을 제거하고 동적인 구성과 더 보기좋은 코드를 구성하기 위함이에요. [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) 는 WebAPI중 하나인데요! WebAPI문서에 들어가서 하나씩 그냥 마음가는대로 읽어보시면 생각보다 많은 정보를 얻을 수 있어요. 신기한것도 많구요! 
```suggestion
export const fetchCategoryNews = async (category, currentPage, searchKeyword) => {
  const params = new URLSearchParams({
    country: 'us',
    pageSize: '15',
    page: currentPage,
    apiKey: apiKey,
  });

  if (searchKeyword) {
      params.append('q', searchKeyword);
  }
  if (category){ 
      params.append('category', category);
  }
  const url = `/top-headlines?${params.toString()}`;
  const response = await instance.get(url);
  return response;
};

```

Reply:
아 그렇네요 어딘가에 저장해두고 말고의 문제가 아니라 다시보니 저렇게 정리해두었어야했군요.
URLSearchParams라는 도구는 처음 봤어요. 사용하니 깔끔하게 정리되네요!

혹시 if문을 
```javascript
  const params = new URLSearchParams({
    country: 'us',
    pageSize: '15',
    page: currentPage,
    apiKey: apiKey,
    ...(searchKeyword && {q: searchKeyword}),
    ...(category && {category: category}),
  });
```

이렇게 대체하는 것은 어떤가요? 이왕하는거 반복을 줄이고자 하였는데, 오히려 가독성이 떨어질 수도 있겠다는 생각이 들어서요!